python/lsst/jointcal/utils.py

```diff
@@ -0,0 +1,368 @@
+"""
+Plot jointcal results and compare with single-frame procesing.
+
+
+"""
+import collections
+
+import numpy as np
+from lsst.afw.geom import arcseconds
+from astropy import units as u
+
+import lsst.afw.table
+
+
+class JointcalStatistics(object):
+    def __init__(self, match_radius=0.1*arcseconds, flux_limit=100.):
+        """
+        Parameters
+        ----------
+        match_radius : lsst.afw.Angle
+            match sources within this radius for RMS statistics
+        flux_limit : float
+            Signal/Noise (flux/fluxSigma) for sources to be included in the RMS cross-match.
+            100 is a balance between good centroids and enough sources.
+        """
+        self.match_radius = match_radius
+        self.flux_limit = 100
+
+    def compute_rms(self, data_refs, visit_list, reference):
+        """
+        Match all data_refs to compute the RMS, for all detections above self.flux_limit.
+
+        Parameters
+        ----------
+        data_refs : list
+            A list of data refs to do the calculations between.
+        visit_list : list
+            list of visit identifiers to do the catalog merge on.
+        reference : lsst reference catalog
+            reference catalog to do absolute matching against.
+
+        Return
+        ------
+        geom.Angle
+            New relative RMS of the matched sources.
+        geom.Angle
+            New absolute RMS of matched sources.
+        """
+        visits_per_dataRef = [dataRef.dataId['visit'] for dataRef in data_refs]
+
+        def compute(catalogs):
+            """Compute the relative and absolute RMS."""
+            visit_catalogs = self._make_visit_catalogs(catalogs, visits_per_dataRef, visit_list)
+            refcat = visit_catalogs.values()[0]  # use the first catalog as the relative reference catalog
+            relative = self._make_match_dict(refcat, visit_catalogs.values()[1:])
+            absolute = self._make_match_dict(reference, visit_catalogs.values())
+            return visit_catalogs, relative, absolute
+
+        old_cats = [dataRef.get('src') for dataRef in data_refs]
+        visit_catalogs, self.old_relative, self.old_absolute = compute(old_cats)
+
+        # Update coordinates with the new wcs including distortion corrections.
+        new_cats = []
+        for dataRef in data_refs:
+            new_cats.append(dataRef.get('src'))
+            lsst.afw.table.utils.updateSourceCoords(dataRef.get('wcs').getWcs(), new_cats[-1])
+        visistCatalogs, self.new_relative, self.new_absolute = compute(new_cats)
```

Comment:
updateSourceCoords modifies the Coords field in new_cats with the other wcs. `dataRef.get('wcs')` gets the jointcal-produced wcs. 'wcs' is a dataset type particular to jointcal and meas_mosaic, and it should go away once the composite dataset work is done and we have a better way to represent it.